payment/src/models/product.ts

```diff
@@ -90,7 +90,7 @@ productSchema.statics.findByEvent = (event: {
   version: number;
 }) => {
   return Product.findOne({
-    _id: event.id,
+    id: event.id,
```

Comment:
This is also an issue that I was encountered in the past.
The problem is `findOne` query in mongoose need to change from `_id` to `id` for it to works correctly
I am still curious that it might come from this line of codes when I convert it to JSON.

```
export const productSchema = new mongoose.Schema<ProductDoc, ProductModel>(
  {
    title: {
      type: String,
      required: true,
    },
    price: {
      type: Number,
      required: true,
      default: 0,
    },
    userId: { type: String, required: true },
    image: {
      type: String,
      required: true,
    },
    colors: { type: String },
    sizes: { type: String },
    countInStock: {
      type: Number,
      required: true,
      default: 1,
    },
    isReserved: {
      type: Boolean,
      required: true,
      default: false,
    },
  },
  {
    toJSON: {
      transform(doc, ret) {
        ret.id = ret._id;
        delete ret._id;
        delete ret.__v;
      },
    },
    timestamps: true,
  }
);
```